.github/workflows/npm-publish.yaml

```diff
@@ -0,0 +1,31 @@
+name: 'Auto-publish NPM Package'
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  push_to_registry:
+    name: 'Setup, and publish'
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: 'Check out the repo'
+        uses: actions/checkout@v4
+
+      - name: 'Set up Node'
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
```

Comment:
One reason I read to use a specific version, is that, once a new `lts` is released, it could bring unexpected changes. Hopefully we're updating this as node evolves, and this won't be an issue.